cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1675,18 +1709,27 @@ def request_integration():
 
 
 @when_none('endpoint.aws.joined',
-           'endpoint.gcp.joined')
+           'endpoint.gcp.joined',
+           'endpoint.openstack.joined',
+           'endpoint.vsphere.joined',
+           'endpoint.azure.joined')
 @when('kubernetes-master.cloud-request-sent')
 def clear_requested_integration():
```

Comment:
Is the `@when('kubernetes-master.cloud-request-sent')` still appropriate for this handler? It seems like we might want this to get called when departing from openstack or vsphere, where `kubernetes-master.cloud-request-sent` never gets set.

Reply:
This stands out as a problem for vsphere in particular. I don't see any way for `kubernetes-master.cloud.blocked` to get cleared after relating to vsphere.

Reply:
Yeah, I changed that in the worker and forgot to do the same for the master.